cli/tests/test_acls.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::Path;
+
+use crate::common::{get_stderr_string, get_stdout_string, TestEnvironment};
+
+fn switch_to_secret_backend(repo_path: &Path) {
+    std::fs::write(
+        repo_path
+            .join(".jj")
+            .join("repo")
+            .join("store")
+            .join("type"),
+        "secret",
+    )
+    .unwrap();
+}
```

Comment:
along with the other comment, could this use the official API instead of going through implementation details.

Reply:
We don't really have any APIs for changing the backend type of an existing repo. I agree that this is a hack but I couldn't think of a better solution. The tests create a repo with some secret data while using the git backend (which of course doesn't care about secret data) and then switch to the secret backend so the ACLs are enforced.

Reply:
I'm in favor of leaving a TODO for a `#[cfg(test)]` API but it fine as is.